lib/src/working_copy.rs

```diff
@@ -867,8 +867,16 @@ impl TreeState {
 
             // TODO: Check that the file has not changed before overwriting/removing it.
             match diff {
+                Diff::Removed(TreeValue::GitSubmodule(_old_id)) => {
+                    println!("warning: git submodule not removed at {:?}", path);
+                    self.file_states.remove(&path);
+                    stats.removed_files += 1;
```

Comment:
I chose to increment the stat here since we also update the `updated_files` counter even though we ignore submodules